parlai/agents/seq2seq/seq2seq.py

```diff
@@ -382,12 +381,13 @@ def eval_step(self, batch):
             self.metrics['num_tokens'] += target_tokens
 
         cand_choices = None
-        if cand_preds is not None:
+        if cand_scores is not None:
+            cand_preds = cand_scores.sort(1, True)[1]
             # now select the text of the cands based on their scores
-            cand_choices = self._pick_cands(cand_preds, cand_inds,
+            cand_choices = self._pick_cands(cand_preds, cand_params[1],
                                             batch.candidates)
 
-        text = [self._v2t(p) for p in preds]
+        text = [self._v2t(p) for p in scores.max(2)[1].cpu()]
```

Comment:
IMHO hiding this max() call in here is a bit nefarious.

```python
_, preds = scores.max(2)
text = [self._v2t(p) for p in preds.cpu()]
```

Reply:
actually introduced a bug too because I used the teacher-forced scores on accident from the if block, I'll update